src/pmcl3d.c

```diff
@@ -584,7 +583,9 @@ int main( int argc, char **argv ) {
 
     if( err ) {
       fprintf( stderr, "Source initialization failed!\n" );
-      return -1;
+
+      MPI_Finalize();
+      return 0;
```

Comment:
`0` is success, that's probably not what we want..

Consider using macros: https://www.gnu.org/software/libc/manual/html_node/Exit-Status.html